Src/BootCamp.Chapter/Inventory.cs

```diff
@@ -15,12 +17,49 @@ public Inventory()
 
         public Item[] GetItems(string name)
         {
-            return new Item[0];
+            Item[] allItems = GetItems();
+            Item[] filteredItems = new Item[1];
+
+            if (allItems[0] == null) return filteredItems;
+
+            foreach (Item Item in allItems)
+            {
+                if (Item.GetName() == name)
+                {
+                    if (allItems.Length > 1)
+                    {
+                        var temp = (Item[]) filteredItems.Clone();
+                        filteredItems = new Item[filteredItems.Length + 1];
+                        temp.CopyTo(filteredItems, 0);
+                    }
+                    filteredItems[^1] = Item;
+                }
+            }
+            return filteredItems;
         }
 
-        public void AddItem(Item item)
+        public Item GetFirstItem(string name)
         {
+            foreach (Item item in _items)
+            {
+                if (item.GetName() == name) return item;
+            }
 
+            return null;
+        }
+
+        public void AddItem(Item item)
+        {
+            if (item == null) return;
+            if (GetItems().Length == 1)
+            {
+                _items[0] = item;
+                return;
+            }
+            Item[] temp = GetItems();
+            _items = new Item[temp.Length + 1];
+            temp.CopyTo(_items, 0);
+            _items[^1] = item;
         }
```

Comment:
Consider an implementation using `Array.Resize()`. 

As an example, `Array.Resize(ref _items, _items.Length + 1)` copies elements to a new array and resizes it to `_items.Length + 1` for you.